atc/wrappa/reject_archived_wrappa_test.go

```diff
@@ -0,0 +1,135 @@
+package wrappa_test
+
+import (
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/atc/db/dbfakes"
+	"github.com/concourse/concourse/atc/wrappa"
+	"github.com/tedsuo/rata"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("RejectArchivedWrappa", func() {
+	var (
+		raWrappa         *wrappa.RejectArchivedWrappa
+		raHandlerFactory wrappa.RejectArchivedHandlerFactory
+	)
+
+	BeforeEach(func() {
+		fakeTeamFactory := new(dbfakes.FakeTeamFactory)
+		raHandlerFactory = wrappa.NewRejectArchivedHandlerFactory(fakeTeamFactory)
+		raWrappa = wrappa.NewRejectArchivedWrappa(raHandlerFactory)
+	})
+
+	It("wraps endpoints", func() {
+		inputHandlers := rata.Handlers{}
+
+		for _, route := range atc.Routes {
+			inputHandlers[route.Name] = &stupidHandler{}
+		}
+
+		expectedHandlers := rata.Handlers{
+
+			atc.PausePipeline:           raHandlerFactory.RejectArchived(inputHandlers[atc.PausePipeline]),
+			atc.CreateJobBuild:          raHandlerFactory.RejectArchived(inputHandlers[atc.CreateJobBuild]),
+			atc.ScheduleJob:             raHandlerFactory.RejectArchived(inputHandlers[atc.ScheduleJob]),
+			atc.CheckResource:           raHandlerFactory.RejectArchived(inputHandlers[atc.CheckResource]),
+			atc.CheckResourceType:       raHandlerFactory.RejectArchived(inputHandlers[atc.CheckResourceType]),
+			atc.DisableResourceVersion:  raHandlerFactory.RejectArchived(inputHandlers[atc.DisableResourceVersion]),
+			atc.EnableResourceVersion:   raHandlerFactory.RejectArchived(inputHandlers[atc.EnableResourceVersion]),
+			atc.PinResourceVersion:      raHandlerFactory.RejectArchived(inputHandlers[atc.PinResourceVersion]),
+			atc.UnpinResource:           raHandlerFactory.RejectArchived(inputHandlers[atc.UnpinResource]),
+			atc.SetPinCommentOnResource: raHandlerFactory.RejectArchived(inputHandlers[atc.SetPinCommentOnResource]),
+			atc.RerunJobBuild:           raHandlerFactory.RejectArchived(inputHandlers[atc.RerunJobBuild]),
+
+			atc.GetConfig:                     inputHandlers[atc.GetConfig],
```

Comment:
I wonder if there's a way of getting having this map explicitly filled only with those that are `RejectArchived` and the rest coming from a `for ...` :thinking:

Reply:
Wouldn't that just move this list somewhere else? I'm trying to think how that would work along with the `panic` in the `default:` case...

Reply:
Could you do something like this?

```go
inputHandlers := rata.Handlers{}
for _, route := range atc.Routes {
  inputHandlers[route.Name] = &stupidHandler{}
}

rejectArchiveRoutes := []string{
  atc.PausePipeline,
  atc.CreateJobBuild,
  ...
}
rejectArchiveLookup := make(map[string]bool)
for _, name := range rejectArchiveRoutes {
  rejectArchiveLookup[name] = true
}

wrappedHandlers := raWrappa.Wrap(inputHandlers)

for name, handler := range inputHandlers {
  expectedHandler := handler
  if rejectArchiveLookup[name] {
    expectedHandler = raHandlerFactory.RejectArchived(expectedHandler)
  }
  Expect(wrappedHandlers[name]).To(BeIdenticalTo(expectedHandler), "handler is "+name)
}
```

Reply:
oh yeah, dumb me didn't realize this was the test file :face_palm: